ers/sql.go

```diff
@@ -121,11 +122,185 @@ func (rdr *SQLEventReader) Serve() (err error) {
+			} else if ruleType == utils.MetaGreaterThan {
+				if elem == utils.EmptyString {
+					singleCond = fmt.Sprintf(" %s > %s", field, value)
+				} else {
+					singleCond = fmt.Sprintf(" JSON_VALUE(%s, '$.%s') > %s", elem, field, value)
+				}
+			} else if ruleType == utils.MetaLessOrEqual {
+				if elem == utils.EmptyString {
+					singleCond = fmt.Sprintf(" %s <= %s", field, value)
+				} else {
+					singleCond = fmt.Sprintf(" JSON_VALUE(%s, '$.%s') <= %s", elem, field, value)
+				}
+			} else if ruleType == utils.MetaLessThan {
+				if elem == utils.EmptyString {
+					singleCond = fmt.Sprintf(" %s < %s", field, value)
+				} else {
+					singleCond = fmt.Sprintf(" JSON_VALUE(%s, '$.%s') < %s", elem, field, value)
+				}
+			}
+		case utils.MetaPrefix, utils.MetaNotPrefix:
+			if not {
+				if elem == utils.EmptyString {
+					conditions = append(conditions, fmt.Sprintf(" %s NOT LIKE '%s%%'", field, value))
+					continue
+				}
+				conditions = append(conditions, fmt.Sprintf(" JSON_VALUE(%s, '$.%s') NOT LIKE '%s%%'", elem, field, value))
+				continue
+			}
+			if elem == utils.EmptyString {
+				singleCond = fmt.Sprintf(" %s LIKE '%s%%'", field, value)
+			} else {
+				singleCond = fmt.Sprintf(" JSON_VALUE(%s, '$.%s') LIKE '%s%%'", elem, field, value)
+			}
+		case utils.MetaSuffix, utils.MetaNotSuffix:
+			if not {
+				if elem == utils.EmptyString {
+					conditions = append(conditions, fmt.Sprintf(" %s NOT LIKE '%%%s'", field, value))
+					continue
+				}
+				conditions = append(conditions, fmt.Sprintf(" JSON_VALUE(%s, '$.%s') NOT LIKE '%%%s'", elem, field, value))
+				continue
+			}
+			if elem == utils.EmptyString {
+				singleCond = fmt.Sprintf(" %s LIKE '%%%s'", field, value)
+			} else {
+				singleCond = fmt.Sprintf(" JSON_VALUE(%s, '$.%s') LIKE '%%%s'", elem, field, value)
+			}
+		case utils.MetaRegex, utils.MetaNotRegex:
+			if not {
+				if elem == utils.EmptyString {
+					conditions = append(conditions, fmt.Sprintf(" %s NOT REGEXP '%s'", field, value))
+					continue
+				}
+				conditions = append(conditions, fmt.Sprintf(" JSON_VALUE(%s, '$.%s') NOT REGEXP '%s'", elem, field, value))
+				continue
+			}
+			if elem == utils.EmptyString {
+				singleCond = fmt.Sprintf(" %s REGEXP '%s'", field, value)
+			} else {
+				singleCond = fmt.Sprintf(" JSON_VALUE(%s, '$.%s') REGEXP '%s'", elem, field, value)
+			}
+		}
+		conditions = append(conditions, singleCond)
+	}
+	return
+}
+
 func (rdr *SQLEventReader) readLoop(db *gorm.DB, sqlDB io.Closer) {
 	defer sqlDB.Close()
 	tm := time.NewTimer(0)
+	var filters []*engine.Filter
+	var whereQueries []string
+	var renewedFltrs []string
+	for _, fltr := range rdr.Config().Filters {
+		if result, err := rdr.dm.GetFilter(config.CgrConfig().GeneralCfg().DefaultTenant, fltr, true, false, utils.NonTransactional); err != nil {
+			rdr.rdrErr <- err
+			return
+		} else {
+			filters = append(filters, result)
+			if !strings.Contains(fltr, utils.DynamicDataPrefix+utils.MetaReq) {
```

Comment:
You should only look for prefix of the element, not contains. Plus you are only checking the inline filters in this way.